include/seqan/blast/blast_statistics.h

```diff
@@ -0,0 +1,1076 @@
+ * @param[in]     value              The new value.
+ * @return isValid(blastScoringScheme) whether Karlin-Altschul values for the new scheme exist.
+ */
+
+inline bool
+setScoreMismatch(BlastScoringScheme<Score<int, Simple>> & scheme, int const val)
+{
+    setScoreMismatch(scheme._internalScheme, val);
+    return _selectSet(scheme);
+}
+
+/*!
+ * @fn BlastScoringScheme#setScoreGapOpenBlast
+ * @headerfile <seqan/blast.h>
+ * @brief Set gap open score in BLAST convention.
+ * @signature bool setScoreGapOpenBlast(blastScoringScheme, value);
+ * @param[in,out] blastScoringScheme The @link BlastScoringScheme @endlink to modify.
+ * @param[in]     value              The new value.
+ * @return isValid(blastScoringScheme) whether Karlin-Altschul values for the new scheme exist.
+ */
+
+template <typename TScore>
+inline bool
+setScoreGapOpenBlast(BlastScoringScheme<TScore> & scheme, typename Value<TScore>::Type const val)
+{
+    scheme._gapOpenBlast = val;
+    setScoreGapOpen(scheme._internalScheme, scoreGapOpenBlast(scheme) + scoreGapExtend(scheme));
+
+    return _selectSet(scheme);
+}
+
+/*!
+ * @fn BlastScoringScheme#setScoreGapOpen
+ * @brief Set gap open score in SeqAn convention.
+ * @signature bool setScoreGapOpen(blastScoringScheme, value);
+ * @param[in,out] blastScoringScheme The @link BlastScoringScheme @endlink to modify.
+ * @param[in]     value              The new value.
+ * @return isValid(blastScoringScheme) whether Karlin-Altschul values for the new scheme exist.
+ */
+
+template <typename TScore>
+inline bool
+setScoreGapOpen(BlastScoringScheme<TScore> & scheme, typename Value<TScore>::Type const val)
+{
+    return setScoreGapOpenBlast(scheme, val - scoreGapExtend(scheme));
+}
+
+/*!
+ * @fn BlastScoringScheme#setScoreGapExtend
+ * @brief Set gap extend score.
+ * @signature bool setScoreGapExtend(blastScoringScheme, value);
+ * @param[in,out] blastScoringScheme The @link BlastScoringScheme @endlink to modify.
+ * @param[in]     value              The new value.
+ * @return isValid(blastScoringScheme) whether Karlin-Altschul values for the new scheme exist.
+ */
+
+template <typename TScore>
+inline bool
+setScoreGapExtend(BlastScoringScheme<TScore> & scheme, typename Value<TScore>::Type const val)
+{
+    setScoreGapExtend(scheme._internalScheme, val);
+    setScoreGapOpen(scheme._internalScheme, scoreGapOpenBlast(scheme) + scoreGapExtend(scheme));
+
+    return _selectSet(scheme);
+}
+
+// ----------------------------------------------------------------------------
+// Function getLambda
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn BlastScoringScheme#getLambda
+ * @headerfile <seqan/blast.h>
+ * @brief Get the &lambda; value of the Karlin-Altschul parameters.
+ * @signature double getLambda(blastScoringScheme);
+ */
+
+template <typename TMatrixSpec>
+inline double
+getLambda(BlastScoringScheme<Score<int, ScoreMatrix<AminoAcid, TMatrixSpec>>> const & scoringScheme)
```

Comment:
Why do you have **two** versions of all these getters for ScoreMatrix and Simple?